tidy3d/components/device/simulation.py

```diff
@@ -33,16 +35,25 @@
 
 from ...log import log
 
-HEAT_BACK_STRUCTURE_STR = "<<<HEAT_BACKGROUND_STRUCTURE>>>"
+DEVICE_BACK_STRUCTURE_STR = "<<<DEVICE_BACKGROUND_STRUCTURE>>>"
 
+HeatBCTypes = (TemperatureBC, HeatFluxBC, ConvectionBC)
+HeatSourceTypes = (UniformHeatSource, HeatSource, HeatFromElectricSource)
+ElectricBCTypes = (VoltageBC, CurrentBC, InsulatingBC)
 
-class HeatSimulation(AbstractSimulation):
-    """Contains all information about heat simulation.
+
+class DeviceSimulationType(str, Enum):
+    HEAT = "HEAT"
+    CONDUCTION = "CONDUCTION"
+
+
+class DeviceSimulation(AbstractSimulation):
```

Comment:
We should discuss the naming.. I'm sorry to bring it up this late in the development.. Mostly want to understand and make sure we are future-proof in the naming.

So my impression is that we have a single "device" (our structures) and then we want to apply simulations on top of it. So maybe we could categorize the types fo simulation by the physics we want to explore, eg. "OpticalSimulation", "HeatSimulation", "ChargeSimulation".

It seems like "DeviceSimulation" is so far capturing the physics "heat" and "charge" but what seems not quite right to me is that it would make sense for an "optical" simulation to be of my device as well?

We're also throwing in EME, ModeSolver, RF simulations, so I guess how do I make sense of the name "device simulation" in the context of all of these other "simulations" and ones that we might want to introduce?

Would it make sense to consider something like "ActiveDeviceSimulation" or "MultiPhysicsSimulation" already?

Maybe let's discuss a bit

@dbochkov-flexcompute